### PR TITLE
feat(common): add owner id encoding for the token-ownership table

### PR DESCRIPTION
**Pull request by YaweiZhang-930**
**Detailed Description**
- Adds a new package under `common/` holding the `owner_id` encoding
- Changed placeholder values in `nosqlplugin/cassandra/semaphore_tokens.go`
- Added `emptyWorkflowID` to the reserved-UUID constants
- Comment corrections in `versioned/v0.51/semaphore_tokens.cql`. 

**Impact Analysis**
No code path writes to `semaphore_tokens` table today. 
- **Backward Compatibility**: No impact. If rows did exist, a newer build's `IF holder = <new value>` guard would not match one holding `__FREE__`, leaving that slot permanently unavailable — but none do.
- **Forward Compatibility**: No impact. An older build would read the new placeholder as a real holder and treat every slot as taken, but there are no rows and no writers on either side.

**Testing Plan**
- **Unit Tests**: Yes. Two fuzz targets covering both round-trip
  directions — `Owner → string → Owner`, and `string → Owner → string`. On the Cassandra side, a test ties each stored placeholder to the encoding of its parts, so a later change to the encoding fails rather than leaving the hardcoded literals stale; existing query-shape tests now reference the constants instead of the old literals.
- **Persistence Tests**: Covered by the existing `SemaphoreTokenPersistenceSuite` which drives seed → grant →
  release against real Cassandra and asserts an unheld slot reads back empty and a granted one reads back as the owner — exercising both placeholders through the real conditional-write path.
- **Integration Tests**: None applicable. The feature has no RPC or service wiring yet, so there
  is no end-to-end path.
- **Compatibility Tests**: None. 

**Rollout Plan**

- **Rollout**: Ordinary deployment, no special steps. No component reads or writes the table, so the change is inert at runtime.
- **Order of deployment**: Does not matter — there is no producer/consumer pair to sequence.
- **Rollback**: Safe, and order does not matter. Rolling back restores the old constants; with no rows in either direction, nothing is left in a state the other build cannot read.
- **Kill switch**: None, and none needed — no runtime path reaches this code.
- **Constraint**: this must land *before* the feature is wired up and any cluster seeds buckets. Once live rows exist, the placeholder is durable data and changing it needs a migration rather than a constant edit.

### CODE REVIEW

File: schema/cassandra/cadence/versioned/v0.51/semaphore_tokens.cql [hunk 1]
```diff
@@ -1,28 +1,29 @@
--- Distributed semaphore: per-token ownership with an inline reverse index.
+-- Distributed semaphore: which run holds which token slot.
```
Comment (YaweiZhang-930) [inline schema/cassandra/cadence/versioned/v0.51/semaphore_tokens.cql]: I made some changes to the comments only.

Comment (YaweiZhang-930) [reply schema/cassandra/cadence/versioned/v0.51/semaphore_tokens.cql]: I guess this change to an older version is forbidden. should I create a new version v0.53?

File: common/semaphore/owner.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2025 Uber Technologies, Inc.
```
Comment (c-warren) [inline common/semaphore/owner.go]: NIT: We don't need copyrights anymore

File: common/semaphore/owner_test.go [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+// encodes to a string that parses back into the same Owner. Round-tripping for all inputs
+// also proves the encoding is injective, since two Owners sharing an encoding could not both
+// come back out of it. The table above pins specific cases; this covers the ones nobody
+// thought to write down.
+func FuzzOwnerStringRoundTrip(f *testing.F) {
+	f.Add("wf-1", "run-1", int64(7))
+	f.Add("a:b", ":c:", int64(-9))
+	f.Add("", "", int64(0))
+	f.Add(":::", "run-1", int64(9223372036854775807))
+	f.Add("12:xy", "3f2504e0-4f89-11d3-9a0c-0305e82c3301", int64(-9223372036854775808))
```
Comment (c-warren) [inline common/semaphore/owner_test.go]: Do we need to provide values here? I think if we just do a bunch of fuzz cases (like the mapper fuzz tests, which do 100 - you can check out `func RunMapperFuzzTest[TInternal any, TExternal any](` for inspiration. 

The point of the fuzz tests is to randomly generate data so that we can find edge cases that we haven't thought - at the cost of extra test duration + non-deterministic tests.

File: common/persistence/nosql/nosqlplugin/cassandra/semaphore_tokens.go [hunk 4]
```diff
@@ -34,14 +34,19 @@ import (
...
+// cannot produce. The -1 and -2 carry no such guarantee; they only keep the two apart.
+//
+// The "0:" is the encoding's length prefix for the empty workflow id, not padding.
 const (
 	emptyTokenID = -1 // token_id on owner rows (key); negative, never a real slot id
 
-	ownerNoneSentinel = "__NONE__" // owner_id on token rows (key)
-	freeSentinel      = "__FREE__" // holder of an unheld token row (LWT-compared)
+	ownerNoneSentinel = "0::30000000-0000-f000-f000-000000000000:-1" // owner_id on token rows (key)
+	freeSentinel      = "0::30000000-0000-f000-f000-000000000000:-2" // holder of an unheld token row (LWT-compared)
```
Comment (c-warren) [inline common/persistence/nosql/nosqlplugin/cassandra/semaphore_tokens.go:49]: NIT: I agree with gitar that computing these from the real constants is safer than making them consts + adding a unit test. The unit test can be deleted, computing it from const will always be correct.

**Review by c-warren — COMMENTED** [review]
Looks good to me!

File: common/semaphore/owner.go (hunk 2 above)
Comment (YaweiZhang-930) [reply common/semaphore/owner.go]: Why?

File: common/semaphore/owner_test.go (hunk 3 above)
Comment (YaweiZhang-930) [reply common/semaphore/owner_test.go]: Good call!

File: common/semaphore/owner.go (hunk 2 above)
Comment (c-warren) [reply common/semaphore/owner.go]: https://github.com/cadence-workflow/cadence/issues/8483 - we switched to apache 2.0 and don't need to have the license header at the top of each file anymore.

File: common/persistence/nosql/nosqlplugin/cassandra/semaphore_tokens.go (hunk 4 above)
Comment (YaweiZhang-930) [reply common/persistence/nosql/nosqlplugin/cassandra/semaphore_tokens.go:49]: The reason I keep these as literals, bc they aren't derived constants, they're fixed bytes in Cassandra: ownerNoneSentinel is part of the token row's primary key.

Computing them means an encoding change silently re-points production at rows that don't exist, with no error. The literal keeps production on what's actually on disk, and only the test fails.

File: common/semaphore/owner_test.go [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+	seed := time.Now().UnixNano()
+	r := rand.New(rand.NewSource(seed))
+	defer func() {
+		if t.Failed() {
+			t.Logf("random seed: %d", seed)
+		}
+	}()
+
+	for i := 0; i < randomIterations; i++ {
+		owner := randomOwner(r)
```
Comment (c-warren) [inline common/semaphore/owner_test.go]: I was imagining that we'd let the fuzzer library do the fuzzing. In my original suggestion it was to avoid having precomputed examples and let the fuzzer do pure fuzzing (e.g allowing nil values, edge case values). 

The owner will be hard, and I get if you need to provide a custom function for that. What I'd do there is something like:
```
testutils.WithCustomFuncs(func(e *types.WorkflowID, c fuzz.Continue) {
  workflowID := // fuzz a string
  workflowIDLen := len(workflowID)
  runID := // fuzz a uuid/string
  ownerID := // fuzz a string
  return fmt.Sprintf("%d:%s:%s:%s", workflowIDLen, workflowID, runID, ownerID)
}
```
That way you generate random data for the workflowID/runID/ownerID without having to write custom functionality to produce something truly random.

**Review by c-warren — APPROVED** [review]
Looks good - I would rather use our existing helpers for the fuzz tests than roll new ones.
